apis/clusters/v1beta1/kafka_webhook_test.go

```diff
@@ -0,0 +1,227 @@
+
+			TestKafkaManifest.Spec.PCICompliance = !kafkaManifest.Spec.PCICompliance
+			Expect(k8sClient.Patch(ctx, &TestKafkaManifest, patch)).ShouldNot(Succeed())
+			TestKafkaManifest.Spec.PCICompliance = kafkaManifest.Spec.PCICompliance
+
+			TestKafkaManifest.Spec.PrivateNetworkCluster = !kafkaManifest.Spec.PrivateNetworkCluster
+			Expect(k8sClient.Patch(ctx, &TestKafkaManifest, patch)).ShouldNot(Succeed())
+			TestKafkaManifest.Spec.PrivateNetworkCluster = kafkaManifest.Spec.PrivateNetworkCluster
+
+			TestKafkaManifest.Spec.PartitionsNumber += 1
+			Expect(k8sClient.Patch(ctx, &TestKafkaManifest, patch)).ShouldNot(Succeed())
+			TestKafkaManifest.Spec.PartitionsNumber = kafkaManifest.Spec.PartitionsNumber
+
+			TestKafkaManifest.Spec.AllowDeleteTopics = !kafkaManifest.Spec.AllowDeleteTopics
+			Expect(k8sClient.Patch(ctx, &TestKafkaManifest, patch)).ShouldNot(Succeed())
+			TestKafkaManifest.Spec.AllowDeleteTopics = kafkaManifest.Spec.AllowDeleteTopics
+
+			TestKafkaManifest.Spec.AutoCreateTopics = !kafkaManifest.Spec.AutoCreateTopics
+			Expect(k8sClient.Patch(ctx, &TestKafkaManifest, patch)).ShouldNot(Succeed())
+			TestKafkaManifest.Spec.AutoCreateTopics = kafkaManifest.Spec.AutoCreateTopics
+
+			TestKafkaManifest.Spec.ClientToClusterEncryption = !kafkaManifest.Spec.ClientToClusterEncryption
+			Expect(k8sClient.Patch(ctx, &TestKafkaManifest, patch)).ShouldNot(Succeed())
+			TestKafkaManifest.Spec.ClientToClusterEncryption = kafkaManifest.Spec.ClientToClusterEncryption
+
+			TestKafkaManifest.Spec.BundledUseOnly = !kafkaManifest.Spec.BundledUseOnly
+			Expect(k8sClient.Patch(ctx, &TestKafkaManifest, patch)).ShouldNot(Succeed())
+			TestKafkaManifest.Spec.BundledUseOnly = kafkaManifest.Spec.BundledUseOnly
+
+			TestKafkaManifest.Spec.PrivateNetworkCluster = !kafkaManifest.Spec.PrivateNetworkCluster
+			Expect(k8sClient.Patch(ctx, &TestKafkaManifest, patch)).ShouldNot(Succeed())
+			TestKafkaManifest.Spec.PrivateNetworkCluster = kafkaManifest.Spec.PrivateNetworkCluster
+
+			TestKafkaManifest.Spec.ClientBrokerAuthWithMTLS = !kafkaManifest.Spec.ClientBrokerAuthWithMTLS
+			Expect(k8sClient.Patch(ctx, &TestKafkaManifest, patch)).ShouldNot(Succeed())
+			TestKafkaManifest.Spec.ClientBrokerAuthWithMTLS = kafkaManifest.Spec.ClientBrokerAuthWithMTLS
+
+			prevSchemaRegistry := kafkaManifest.Spec.SchemaRegistry
+			TestKafkaManifest.Spec.SchemaRegistry = []*SchemaRegistry{prevSchemaRegistry[0], prevSchemaRegistry[0]}
+			Expect(k8sClient.Patch(ctx, &TestKafkaManifest, patch)).ShouldNot(Succeed())
+			TestKafkaManifest.Spec.SchemaRegistry = prevSchemaRegistry
+
+			prevKarapaceSchemaRegistry := kafkaManifest.Spec.KarapaceSchemaRegistry
+			TestKafkaManifest.Spec.KarapaceSchemaRegistry = []*KarapaceSchemaRegistry{prevKarapaceSchemaRegistry[0], prevKarapaceSchemaRegistry[0]}
+			Expect(k8sClient.Patch(ctx, &TestKafkaManifest, patch)).ShouldNot(Succeed())
+			TestKafkaManifest.Spec.KarapaceSchemaRegistry = prevKarapaceSchemaRegistry
```

Comment:
Should we also test that the values itself were not changed?

Reply:
discussed